src/main/java/com/exadel/discount/repository/UserRepository.java

```diff
@@ -0,0 +1,12 @@
+package com.exadel.discount.repository;
+
+import com.exadel.discount.entity.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.UUID;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, UUID> {
+    User findByEmail(String email);
```

Comment:
change return type to optional

Reply:
This is done locally.

Reply:
It was resolved in commit "Commit with amendments №2".